modules/phase_field/include/utils/EBSimpleTrees.h

```diff
@@ -0,0 +1,53 @@
+
+#pragma once
+
+#include "ExpressionBuilderToo.h"
+
+struct SimplificationRules
+{
+  SimplificationRules(std::vector<std::vector<ExpressionBuilderToo::EBTerm>> rules)
+  {
+    for (auto rule : rules)
+    {
+      _simplification_trees.emplace_back(
+          std::make_pair(rule[0].cloneRoot()->toNNary(), rule[1].cloneRoot()->toNNary()));
+    }
+  }
+  operator std::vector<
+      std::pair<ExpressionBuilderToo::EBTermNode *, ExpressionBuilderToo::EBTermNode *>>()
+  {
+    return _simplification_trees;
+  }
+  std::vector<std::pair<ExpressionBuilderToo::EBTermNode *, ExpressionBuilderToo::EBTermNode *>>
+      _simplification_trees;
+};
+
+class PrepEBSimpleTrees
+{
+public:
+  PrepEBSimpleTrees();
+
+  operator std::vector<
+      std::pair<ExpressionBuilderToo::EBTermNode *, ExpressionBuilderToo::EBTermNode *>>()
+  {
+    return _prep_simplification_trees;
+  }
+
+  std::vector<std::pair<ExpressionBuilderToo::EBTermNode *, ExpressionBuilderToo::EBTermNode *>>
+      _prep_simplification_trees;
+};
+
+class EBSimpleTrees
```

Comment:
`EBSimpleTrees` looks identical to `PrepEBSimpleTrees`

Reply:
They are very similar, but the PrepEBSimpleTrees has some rules that set up for changing the additions to EBNNaryOpTermNodes.

Reply:
You don't need a separate class for that, only a separate instantiation.

Reply:
Would it be appropriate to have two constructors, but one with a dummy bool variable to distinguish between the two?

Reply:
I don't understand where you need to make that distinction.

Reply:
My mistake, I fixed it in the last update.